content/HTTPS is not enough - Pablo Blanco.md

```diff
@@ -0,0 +1,218 @@
+
+## How can we mitigate HTTPS vulnerabilities?
+
+Whenever a browser requests a page from a web server, the server answers to the
+browser with the content along with some useful information, contained in
+**headers**. Many SSL-related vulnerabilities can be mitigated with a special
+type of headers, called security headers that can tell browsers how to treat
+content on a web page such as “whitelist certain types of content from trusted
+sites” or “only load this site over HTTPS”, for this particular vulnerabilities
+there are 2 types of headers we can use, **HSTS** and **CSP**.
+
+### HTTP Strict Transport Security (HSTS)
+
+The HTTP Strict-Transport-Security response header, lets a website tell browsers
+that it should only be accessed using HTTPS strictly, instead of using HTTP and
+should automatically convert all attempts to access the site using HTTP to HTTPS
+requests instead.
+
+This header is specified by [RFC6797](https://tools.ietf.org/html/rfc6797) and
+contains three directives:
+
+- **max-age** (mandatory): time in seconds, that the browser should remember
+  that a site is only to be accessed using HTTPS.
+
+- **includesubdomains** (optional): indicates if this rule applies to all of the
+  site's subdomains as well.
+
+- **preload** (optional): indicates that the site meets requirements for HSTS
+  preloading and is on the HSTS preload list or has applied for it.
+
+Once the browser has set up HSTS for a site, all communication with that site
+will use HTTPS, which **eliminates the SSL stripping threat** and guarantees
+secure transmission.
+
+#### Keep browsers updated
+
+Everything looks fine still here, but the first time an user visits an unknown
+site, the browser relies on the server’s responses to determine the correct
+protocol. Attackers could intercept this first visit and redirect it to an
+insecure website using a man-in-the-middle attack. This is not as rare as it
+seems – it just means that the site is not present in the browser’s HSTS
+database. While this can be the result of a time-based attack via NTP, it can
+also happen if the max-age expiry period since the last visit has passed, the
+browser is a fresh install, the server is misconfigured (for example max-age was
+left at 0 after testing) and so on.
+
+To remedy this issue and also improve page load speeds,
+[modern browsers](https://caniuse.com/#feat=stricttransportsecurity) (Chrome,
+Firefox, Safari, among others) check if a certain domain is HSTS-enabled against
+a common preloaded list called [hstspreload.org](https://hstspreload.org/).
+
+There, webmasters and site administrators can submit it’s domains for inclusion
+in Chrome's HTTP Strict Transport Security (HSTS) preload list. By including our
+domains to that list we are sure to force the user’s browser to navigate our
+site over HTTPS **strictly**.
+
+If a site is listed, the browser always connects to it using HTTPS, even on the
```

Comment:
```suggestion
If a site is listed, the browser always connect to it using HTTPS, even on the
```

Reply:
Seba, here it should be `connects`, because the browser refers to `it`, and `he`, `she` and `it` in the present tense should go with `s`

Reply:
true dat! sorry, the `s` in always sometimes throws me off :D